jwst/lib/set_velocity_aberration.py

```diff
@@ -45,95 +45,50 @@ def aberration_scale(velocity_x, velocity_y, velocity_z,
-        should be the same as the coordinate system for the velocity.
+        point, such as the center of a detector) in the barycentric coordinate
+        system.  The equator and equinox should be the same as the coordinate
+        system for the velocity.
 
     Returns
     -------
     scale_factor: float
         Multiply the nominal image scale (e.g. in degrees per pixel) by
         this value to obtain the image scale corrected for the "aberration
         of starlight" due to the velocity of JWST with respect to the Sun.
-    """
-
-    speed = math.sqrt(velocity_x**2 + velocity_y**2 + velocity_z**2)
-    if speed == 0.0:
-        logger.warning('Speed is zero. Forcing scale to 1.0')
-        return 1.0
-
-    beta = speed / SPEED_OF_LIGHT
-    gamma = 1. / math.sqrt(1. - beta**2)
-
-    # [targ_x, targ_y, targ_z] is a unit vector.
-    r_xy = math.cos(targ_dec * d_to_r)          # radial distance in xy-plane
-    targ_x = r_xy * math.cos(targ_ra * d_to_r)
-    targ_y = r_xy * math.sin(targ_ra * d_to_r)
-    targ_z = math.sin(targ_dec * d_to_r)
 
-    dot_prod = (velocity_x * targ_x +
-                velocity_y * targ_y +
-                velocity_z * targ_z)
-    cos_theta = dot_prod / speed
-    # This sin_theta is only valid over the range [0, pi], but so is the
-    # angle between the velocity vector and the direction toward the target.
-    sin_theta = math.sqrt(1. - cos_theta**2)
+    apparent_ra: float
+        Aparent star position in the moving telescope frame.
 
-    tan_theta_p = sin_theta / (gamma * (cos_theta + beta))
-    theta_p = math.atan(tan_theta_p)
+    apparent_dec: float
+        Aparent star position in the moving telescope frame.
 
-    scale_factor = (gamma * (cos_theta + beta)**2 /
-                    (math.cos(theta_p)**2 * (1. + beta * cos_theta)))
-
-    return scale_factor
-
-
-def aberration_offset(velocity_x, velocity_y, velocity_z,
-                      targ_ra, targ_dec):
-    """Compute the RA/Dec offsets due to velocity aberration.
-
-    Parameters
-    ----------
-    velocity_x, velocity_y, velocity_z: float
-        The components of the velocity of JWST, in km / s with respect to
-        the Sun.  These are celestial coordinates, with x toward the
-        vernal equinox, y toward right ascension 90 degrees and declination
-        0, z toward the north celestial pole.
-
-    targ_ra, targ_dec: float
-        The right ascension and declination of the target (or some other
-        point, such as the center of a detector).  The equator and equinox
-        should be the same as the coordinate system for the velocity.
-
-    Returns
-    -------
-    delta_ra, delta_dec: float
-        The offset to be added to the input RA/Dec, in units of radians.
     """
+    beta = np.array([velocity_x, velocity_y, velocity_z]) / SPEED_OF_LIGHT
+    beta2 = np.dot(beta, beta)
+    if beta2 == 0.0:
+        return 1.0, targ_ra, targ_dec
```

Comment:
The old version of the code issued a warning when the velocity is zero. That would be good to maintain here, just for traceability in the processing log.

Reply:
The old warning `logger.warning('Speed is zero. Forcing scale to 1.0')` is interesting. When speed is 0 then mathematically scale is 1. There is nothing exceptional here. Saying the code is "forcing" scale to be 1 is kind of a forceful statement :) I think from the caller's perspective it should not matter how the code handles special cases. In fact, this `if` can be safely removed from the code if we are talking only about scale. It is here because of the aberrated RA and Dec - that formula should be taking a limit in mathematical sense when `v->0`. In addition, it is really an extreme check to make code work even when JWST is "launched" in a garage. We do not reaaly expect a flying JWST with speed 0.

However, if you are thinking of detecting case when software incorrectly passes 0 speed instead of `v != 0` - then I hope someone will be reading those log files. Otherwise I would suggest using an `assert` or raising some other exception.

Reply:
Let me know what do you prefer.